### ISSUE TITLE
[YCQL] Track difference of dateof() and totimestamp().

### ISSUE THREAD

**Issue by bbaddepudi** [YugaByte/yugabyte-db#924]
@kmuthukk  wrote:
From YB docs, the difference seems minor (just that one take two input types, and the other only 1 input type). It seems to suggest both return the same type.

1. https://docs.yugabyte.com/latest/api/ycql/function_datetime/#totimestamp
This function converts a date or TIMEUUID to the corresponding timestamp. It takes in an argument of type DATE or TIMEUUID. The return value is a TIMESTAMP.

2. https://docs.yugabyte.com/latest/api/ycql/function_datetime/#dateof
This function converts a TIMEUUID to the corresponding timestamp. It takes in an argument of type TIMEUUID. The return value is a TIMESTAMP.

3. Apache Cassandra's documentation on dateOf() differs based on https://docs.datastax.com/en/cql/3.3/cql/cql_reference/timeuuid_functions_r.html
Used in a SELECT clause, this function extracts the timestamp of a timeuuid column in a result set. This function returns the extracted timestamp as a date. Use unixTimestampOf() to get a raw timestamp.

Need to check if it is a code issue or only doc issue.

**Comment by bbaddepudi** [comment]
From @m-iancu 
Main difference is one of them is deprecated in standard CQL (dateOf() was older and pretty much replaced with the more consistently named toTimestamp()). I think our implementation and documentation are correct (though docs are maybe incomplete/unclear). See: http://cassandra.apache.org/doc/latest/cql/functions.html#time-conversion-functions and http://cassandra.apache.org/doc/latest/cql/changes.html#id10

From @kmuthukk:
 So, basically, I suppose you are saying that the DataStax docs << This function returns the extracted timestamp as a date. >> for DateOf() is incorrect, and dateOf() actually returns a timestamp, and not a date. In which case, maybe not much for us to do. We can just close the github - with this information.

So closing this issue out.
